datatree/datatree.py

```diff
@@ -87,6 +91,7 @@ class DataTree(
     MappedDatasetMethodsMixin,
     MappedDataWithCoords,
     DataTreeArithmeticMixin,
+    AttrAccessMixin,
```

Comment:
We might want to inherit from xarray's `DataWithCoords` here instead, which also inherits from `AttrAccessMixin`.

I'm guilty of building a heavy dependency on xarray's internals here, but this whole package has that problem so :man_shrugging: